persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -134,6 +137,13 @@
          try {
             db = handler.open(getLocation(), dataDbOptions());
             expiredDb = openDatabase(getExpirationLocation(), expiredDbOptions());
+
+            MetadataImpl existingMeta = handler.loadMetadata();
+            if (existingMeta != null) {
+               // TODO Perform validation/migration
+            }
+            // Update the metadata entry to use the current Infinispan version
+            handler.writeMetadata();
```

Comment:
I am not sure we want to do this if the versions don't match.

Reply:
My plan was that if there are any incompatibilies they will be handled via the migration logic between `x` and latest version, or if this is not possible an exception is thrown before this statement is reached.

Reply:
Oh I agree. I am more saying in case we never get this completed. I know how some features have ideas for the future but it never comes to fruition.

Reply:
Ok. I've added a variable to prevent this in the meantime.